src/ray/util/tests/pipe_logger_test.cc

```diff
@@ -132,50 +109,91 @@ TEST(PipeLoggerTestWithTee, RedirectionWithTee) {
 
-  auto log_token = CreateRedirectionFileHandle(logging_option);
-  ASSERT_EQ(write(log_token.GetWriteHandle(), kLogLine1.data(), kLogLine1.length()),
-            kLogLine1.length());
-  ASSERT_EQ(write(log_token.GetWriteHandle(), kLogLine2.data(), kLogLine2.length()),
-            kLogLine2.length());
-  log_token.Close();
+  auto stream_redirection_handle = CreateRedirectionFileHandle(stream_redirection_opt);
+  stream_redirection_handle.CompleteWrite(kLogLine1.data(), kLogLine1.length());
+  stream_redirection_handle.CompleteWrite(kLogLine2.data(), kLogLine2.length());
+  stream_redirection_handle.Close();
 
   // Check content tee-ed to stderr.
   const std::string stderr_content = testing::internal::GetCapturedStderr();
   EXPECT_EQ(stderr_content, absl::StrFormat("%s%s", kLogLine1, kLogLine2));
 
   // Check log content after completion.
-  const std::string log_file_path1 = test_file_path;
   EXPECT_EQ(CompleteReadFile(test_file_path), kLogLine2);
-
-  const std::string log_file_path2 = absl::StrFormat("%s.1", test_file_path);
   EXPECT_EQ(CompleteReadFile(log_file_path2), kLogLine1);
-
-  // Delete temporary file.
-  EXPECT_EQ(unlink(log_file_path1.data()), 0);
-  EXPECT_EQ(unlink(log_file_path2.data()), 0);
 }
 
 }  // namespace
 
 }  // namespace ray
 
 #endif
+
+/////////////////////////////////////////////////
+// Unit test for both windows and unix platform.
+/////////////////////////////////////////////////
+
+namespace ray {
+
+namespace {
+
+class PipeLoggerWithRedirection : public ::testing::TestWithParam<size_t> {};
+
+TEST_P(PipeLoggerWithRedirection, NoPipeWrite) {
+  const size_t pipe_buffer_size = GetParam();
+  setenv(kPipeLogReadBufSizeEnv.data(),
+         absl::StrFormat("%d", pipe_buffer_size).data(),
+         /*overwrite=*/1);
+
+  // TODO(hjiang): We should have a better test util, which allows us to create a
+  // temporary testing directory.
+  const std::string test_file_path = absl::StrFormat("%s.out", GenerateUUIDV4());
+
+  // Delete temporary file.
+  absl::Cleanup cleanup_test_file = [&test_file_path]() {
+    EXPECT_TRUE(std::filesystem::remove(test_file_path));
+  };
+
+  // Take the default option, which doesn't have rotation enabled.
+  StreamRedirectionOption stream_redirection_opt{};
+  stream_redirection_opt.file_path = test_file_path;
+  auto stream_redirection_handle = CreateRedirectionFileHandle(stream_redirection_opt);
+  stream_redirection_handle.CompleteWrite(kLogLine1.data(), kLogLine1.length());
+  stream_redirection_handle.CompleteWrite(kLogLine2.data(), kLogLine2.length());
+  stream_redirection_handle.Close();
+
+  // Check log content after completion.
+  const auto actual_content = CompleteReadFile(test_file_path);
+  const std::string expected_content = absl::StrFormat("%s%s", kLogLine1, kLogLine2);
+  EXPECT_EQ(actual_content, expected_content);
+}
+
+INSTANTIATE_TEST_SUITE_P(PipeLoggerWithRedirection, PipeLoggerWithRedirection, testing::Values(1024, 3));
```

Comment:
I don't think we need to create another test suite which is exactly the same as PipeLoggerTest. 

Concretely we can do

```
class PipeLoggerTest : public ::testing::TestWithParam<size_t> {};


#if defined(__APPLE__) || defined(__linux__)
TEST_P(PipeLoggerTest, A) {...}
TEST_P(PipeLoggerTest, B) {...}
#endif

TEST_P(PipeLoggerTest, NoPipeWrite) {...}

INSTANTIATE_TEST_SUITE_P(PipeLoggerTest, PipeLoggerTest, testing::Values(1024, 3));
```

As we support more things in Windows, we will gradually move tests out of `#if defined(__APPLE__) || defined(__linux__)`

Reply:
updated